src/engine/source/api/include/api/test/sessionManager.hpp

```diff
@@ -0,0 +1,86 @@
+#ifndef _SESSIONS_MANAGER_HPP
+#define _SESSIONS_MANAGER_HPP
+
+#include "iSessionManager.hpp"
+
+#include <shared_mutex>
+#include <unordered_map>
+
+namespace api::sessionManager
+{
+
+class SessionManager : public ISessionManager
+{
+private:
+    std::unordered_map<std::string, Session> m_activeSessions;
+    std::unordered_map<std::string, std::string> m_policyMap;
+    std::unordered_map<std::string, std::string> m_routeMap;
+
+    std::shared_mutex m_sessionMutex;
+
+public:
+    SessionManager() = default;
+    SessionManager(const SessionManager&) = delete;
+    SessionManager& operator=(const SessionManager&) = delete;
+
+    /**
+     * @brief Get a Session Manager instance.
+     *
+     * @return SessionManager&
+     */
+    static SessionManager& getInstance(void);
+
+    /**
+     * @brief Create a new session.
```

Comment:
suggestion: In Doxygen, you can use the @copydoc tag to indicate that the documentation for a class or function is copied from somewhere else, such as the base class or interface. This avoids redundant documentation in the inheriting class.
example:
`/**
 * @class DerivedClass
 * @brief Clase derivada
 *
 * @copydoc BaseClass
 */`